### DISCUSSION TITLE
Is there a way to cancel an optimistic mutation response ?

### DISCUSSION THREAD

**Discussion by strblr** [urql-graphql/urql#1652] [Q&A] [answered]
Hello !

I have a case in a project where graphcache might or might not have enough information to create an optimistic response for a mutation. But in case it does, it should put out an optimistic response. But the `optimistic` callbacks don't seem to be cancellable once called, they have to return a response.

Wouldn't it be a good idea to be able to return `undefined` to signal a last-minute cancellation and to treat the mutation as if there was no optimistic response ?

Best regards.

**Comment by kitten** [comment, marked as answer]
@ostrebler You can totally return `undefined` from an optimistic updater, so that shouldn't cause any issues. You'll have to make sure though that your updater bails when it does encounter these `null` results on their fields.

**Reply by strblr** [reply]
So returning `undefined` doesn't actually cancel the optimistic behavior, but rather sets `null` as the optimistic result of the mutation, right ?
